app/components/img-selector.js

```diff
@@ -15,6 +16,9 @@ export default Ember.Component.extend({
                 }
             };
             reader.readAsDataURL(e.target.files[0]);
+        },
+        clickInput: function() {
+            this.$().find('.img-selector-input').click();
```

Comment:
1. No. This should be only descendants of the component's $element
2. No, it's a class selector
3. I'd rather not make that assumption unnecessarily